FirebaseAuth/Sources/Swift/User/User.swift

```diff
@@ -1754,53 +1717,33 @@ extension User: NSSecureCoding {}
   #endif
 
   // Update the new token and refresh user info again.
-  private func updateTokenAndRefreshUser(idToken: String, refreshToken: String,
-                                         accessToken: String?,
+  private func updateTokenAndRefreshUser(idToken: String,
+                                         refreshToken: String,
                                          expirationDate: Date?,
                                          result: AuthDataResult,
-                                         requestConfiguration: AuthRequestConfiguration,
-                                         completion: ((AuthDataResult?, Error?) -> Void)?,
-                                         withTaskComplete complete: AuthSerialTaskCompletionBlock? =
-                                           nil) {
+                                         requestConfiguration: AuthRequestConfiguration) async throws
+    -> AuthDataResult {
     tokenService = SecureTokenService(
       withRequestConfiguration: requestConfiguration,
       accessToken: idToken,
       accessTokenExpirationDate: expirationDate,
       refreshToken: refreshToken
     )
-    internalGetToken { response, error in
-      if let error {
-        User.callInMainThreadWithAuthDataResultAndError(callback: completion,
-                                                        complete: complete,
-                                                        error: error)
-        return
-      }
-      guard let accessToken else {
-        fatalError("Internal Auth Error: nil access Token")
-      }
-      let getAccountInfoRequest = GetAccountInfoRequest(accessToken: accessToken,
```

Comment:
This should have passed `response` instead of `accessToken`.

The new code rationalizes the parameters and variable names.